controllers/istio/istio_controller_test.go

```diff
@@ -872,6 +872,47 @@ func TestGetPruningGracePeriod(t *testing.T) {
 	}
 }
 
+func TestManagesExternalRevision(t *testing.T) {
+	tests := []struct {
+		name     string
+		spec     v1.IstioSpec
+		expected bool
+	}{
+		{
+			name:     "Empty spec.values does not manage",
```

Comment:
What if `spec.values` is empty, but `spec.profile` points to a profile that sets the `EXTERNAL_ISTIOD` value. I don't think such a profile exist right now, but should we handle this to protect from such profiles in the future? But I don't even know how we should do this exactly, since the profiles are applied within the helm rendering process.

Reply:
> What if spec.values is empty, but spec.profile points to a profile that sets the EXTERNAL_ISTIOD value.

If the profile sets the `EXTERNAL_ISTIOD` value, isn't it already applied to the values at this point by this: https://github.com/istio-ecosystem/sail-operator/blob/84c35a10c7bcd11d9ac95179cb4258385b9e0017/pkg/revision/values.go#L45-L49

Reply:
No. This was how we initially applied the profiles. Now, the profiles are implemented within the helm charts, so the controller doesn't even know what the final values are. But we kept this ApplyProfilesAndPlatform function, because a few fields are still applied the old way.

Reply:
Well then if there was a profile that set `EXTERNAL_ISTIOD: true` explicitly then this patch wouldn't help. Long term the sail operator should be able to determine if the external cluster is in use.